src/core/mac/data_poll_sender.cpp

```diff
@@ -496,8 +536,88 @@ void DataPollSender::HandlePollTimer(Timer &aTimer)
 
 uint32_t DataPollSender::GetDefaultPollPeriod(void) const
 {
-    return Time::SecToMsec(Get<Mle::MleRouter>().GetTimeout()) -
-           static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts;
+    uint32_t period = Time::SecToMsec(Get<Mle::MleRouter>().GetTimeout());
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (Get<Mac::Mac>().IsCslEnabled())
+    {
+        period = OT_MIN(period, Time::SecToMsec(Get<Mac::Mac>().GetCslTimeout()));
+    }
+#endif
+
+    period -= (static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts);
```

Comment:
How about if we protect the calculation here for the roll-over? (if what we subtract is larger than the `period`)?

Reply:
Sounds good!